pkg/kubelet/kubelet_test.go

```diff
@@ -130,6 +130,13 @@ func newTestKubelet(t *testing.T) *TestKubelet {
 		runtimeCache: kubelet.runtimeCache,
 		t:            t,
 	}
+
+	kubelet.prober = prober.FakeProber{}
+	kubelet.probeManager = prober.NewManager(
```

Comment:
Hmm...you are using the real probe manager. Dose that mean it'll launch goroutines/workers for the tests? Not sure if this is desirable here since we don't really test probing here (or do we?). Might be worth adding a fake probe manager and test that correct functions are called. I am okay with leaving this as it is for now since this PR is quite large.